src/pages/kb/user-guide/querying/query-results-data-source.md

```diff
@@ -6,30 +6,30 @@ title: Querying Existing Query Results
 slug: query-results-data-source
 ---
 
-The **Query Results** Data Source allows you to run queries on top of
-existing query results, so you can easily merge results or perform any other
-kind of "post-processing".
+The **Query Results** Data Source lets you run queries against results from your other Data Sources. Use it to join data from multiple databases or perform other kinds post-processing. Redash uses an in-memory SQLite database to make this possible. As a result, queries against large result sets may fail if Redash runs out of memory. 
 
-To use it, you need to create a new data source of type "Query Results". Once
-you've done this you can use it to write queries like:
+### Setup
+You can enable **Query Results** under the `Data Source` tab of the settings menu. Setup is easy: just provide a name for the source. This is the name that will appear in the source dropdown of the query editor.
 
-    SELECT a.name, b.count
-    FROM query_123 a
+### Querying
+[SQLite query syntax](https://sqlite.org/lang.html) should be familiar if you have worked with other SQLs. Here's an example query:
+
+    SELECT a.name, b.count 
+    FROM query_123 a 
     JOIN query_456 b ON a.id = b.id
 
-When referencing a query as a table in the Query Results query, the table name
-is the query ID (the number in the URL) prefixed with _query_. So for
-example, a query with the URL
-`https://app.redash.io/acme/queries/49588/source` will become `query_49588`
-when referenced as a table.
+Each of your existing queries constitutes its own "table" to SQLite. The table name is the string `query_` concatenated with your desired Query ID, which can be found in that query's URL. So for example, a query with the URL `https://app.redash.io/acme/queries/49588/source` will have the table name `query_49588` in SQLite.
 
-{% callout %}
-You need to make sure the table name (`query_...`) is on the same line as the
-`FROM`/`JOIN` keywords.
+{% callout warning %}
+Be careful that your table name (`query_49588` above) appears on the same line as the associated `FROM` and `JOIN` keywords. Redash can't parse it correctly otherwise.
 {% endcallout %}
 
-Few notes:
+### Cached Query Results
+When you query the **Query Results** Data Source, Redash executes the underlying queries first. This ensures you have recent results in the event that you [schedule this query](/help/user-guide/querying/scheduling-a-query). You can reduce the running time of **Query Results** queries by using `cached_query_` for your table names instead of `query_`. This tells Redash to use the cached results from the most recent execution of a given query. This reduces the number of calls to your underlying Data Sources, improving performance by using older data. Here's an example query:
```

Comment:
It's not entirely clear which query is the one you schedule. Worth clarifying that it's "... in the event that you schedule the Query Results query.".